gdal/frmts/grib/gribdataset.cpp

```diff
@@ -1058,6 +1290,36 @@ int GRIBDataset::Identify( GDALOpenInfo *poOpenInfo )
     return FALSE;
 }
 
+/************************************************************************/
+/*                                Inventory()                           */
+/************************************************************************/
+
+std::unique_ptr<gdal::grib::InventoryWrapper> GRIBDataset::Inventory(VSILFILE *fp, GDALOpenInfo *poOpenInfo)
+{
+    std::unique_ptr<gdal::grib::InventoryWrapper> pInventories;
+
+    VSIFSeekL(fp, 0, SEEK_SET);
+    CPLString sSideCarFilename = CPLString(poOpenInfo->pszFilename) + ".idx";
+    VSILFILE *fpSideCar = nullptr;
+    if ((fpSideCar = VSIFOpenL(sSideCarFilename, "rb")) != nullptr) {
```

Comment:
maybe add a USE_IDX=YES/NO open option, and test it here with ``CPLTestBool(CSLFetchNameValueDef(poOpenInfo->papszOpenOptions, "USE_IDX", "YES"))``